validate.py

```diff
@@ -71,12 +71,19 @@ def main(argv):
     if FLAGS.expand_validation_dirs:
         tf_records = []
         with utils.logged_timer("Building lists of holdout files"):
-            for record_dir in validation_paths:
-                tf_records.extend(gfile.Glob(os.path.join(record_dir, '*', '*', '*.zz')))
+            dirs = validation_paths
+            while len(dirs) > 0:
+                d = dirs.pop()
+                for path, newdirs, files in os.walk(d):
+                    tf_records.extend([os.path.join(path, f) for f in files if f.endswith('.zz')])
```

Comment:
using [] inside .extend() is an antipattern because you accumulate the entire extension in memory as a list, and then copy it over to the existing list one by one.

Reply:
whoops, thanks.